extensions/binding/org.eclipse.smarthome.binding.mqtt/src/main/java/org/eclipse/smarthome/binding/mqtt/internal/MqttBrokerHandlerFactory.java

```diff
@@ -46,14 +51,19 @@
  * @author David Graeff - Initial contribution
  */
 @NonNullByDefault
-@Component(immediate = true, service = { ThingHandlerFactory.class,
-        MQTTTopicDiscoveryService.class }, configurationPid = "MqttBrokerHandlerFactory")
-public class MqttBrokerHandlerFactory extends BaseThingHandlerFactory implements MQTTTopicDiscoveryService {
+@Component(immediate = true, service = { ThingHandlerFactory.class, MQTTTopicDiscoveryService.class,
+        MQTTBrokerConnectionDiscoveryService.class }, configurationPid = "MqttBrokerHandlerFactory")
```

Comment:
configurationPids of handler factories are always of the syntax `binding.<bindingid>`, so here probably best something like `binding.mqttbroker`.